ouroboros-consensus/test-util/Test/Util/Orphans/Arbitrary.hs

```diff
@@ -111,6 +129,129 @@ instance Arbitrary Seed where
 
     shrink = const []
 
+{-------------------------------------------------------------------------------
+  ConnectionPlan
+-------------------------------------------------------------------------------}
+
+-- | Inclusive bounds for each join delay in @'arbitrary' :: 'ConnectionPlan'@.
+boundsConnectionPlanDelay :: (Int, Int)
+boundsConnectionPlanDelay = (0, n - 1)
+{-
+
+  TODO how to explain/justify upper bound? Should it be higher?
+
+  n - 1 BFT passed 10000 tests
+
+  n BFT consistenly fails with at least the following
+
+      *** Failed! (after 560 tests and 1 shrink):
```

Comment:
Wow, that's a strange error. I'll investigate this.

Reply:
I found some more like this. See the `Test.Dynamic.BFT` module as of my Tuesday push.

Reply:
Resolved by #810 and #812